packages/wonder-blocks-timing/src/hooks/use-interval.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+import {useEffect, useState, useCallback} from "react";
+
+import {
+    SchedulePolicy as SchedulePolicies,
+    ClearPolicy as ClearPolicies,
+} from "../util/policies.js";
+import type {IInterval, ClearPolicy, Options} from "../util/types.js";
+
+import {useMountedRef} from "./internal/use-mounted-ref.js";
+import {useUpdatingRef} from "./internal/use-updating-ref.js";
+import {useSimpleInterval} from "./internal/use-simple-interval.js";
+
+export function useInterval(
+    action: () => mixed,
+    intervalMs: number,
+    options?: Options,
+): IInterval {
+    if (typeof action !== "function") {
+        throw new Error("Action must be a function");
+    }
+
+    if (intervalMs < 1) {
+        throw new Error("Interval period must be >= 1");
+    }
+
+    const schedulePolicy =
+        options?.schedulePolicy ?? SchedulePolicies.Immediately;
+
+    const [isSet, setIsSet] = useState(
+        schedulePolicy === SchedulePolicies.Immediately,
+    );
+
+    const set = useCallback(() => setIsSet(true), []);
+
+    const actionRef = useUpdatingRef(action);
+
+    const clear = useCallback(
+        (policy?: ClearPolicy) => {
+            policy = policy ?? options?.clearPolicy;
+            if (isSet && policy === ClearPolicies.Resolve) {
+                actionRef.current();
+            }
+            setIsSet(false);
+        },
+        // react-hooks/exhaustive-deps doesn't require refs to be
```

Comment:
Bummer... seems like there might be a chance this will be addressed if this lands
https://github.com/facebook/react/pull/20477